src/logging.rs

```diff
@@ -1,8 +1,55 @@
 use std::{borrow::Cow, str::FromStr};
 
 use sentry::types::Dsn;
+use serde::{Deserialize, Serialize};
+use tracing::level_filters::LevelFilter;
 
-use crate::config::Config;
+use crate::config;
+
+use tracing_subscriber::{prelude::*, Layer};
+
+/// Controls the log format.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, Deserialize, Serialize, clap::ValueEnum)]
+#[serde(rename_all = "lowercase")]
+pub enum LogFormat {
+    /// Auto detect the best format.
+    Auto,
+
+    /// Pretty printing with colors.
+    Pretty,
+
+    /// Simplified plain text output.
+    Simplified,
+
+    /// Dump out JSON lines.
+    Json,
+}
+
+/// The logging level
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, Deserialize, Serialize, clap::ValueEnum)]
+#[serde(rename_all = "lowercase")]
+pub enum Level {
+    Error,
+    Warn,
+    Info,
+    Debug,
+    Trace,
+    Off,
+}
+
+impl Level {
+    /// Returns the tracing [`LevelFilter`].
+    pub const fn level_filter(&self) -> LevelFilter {
+        match self {
+            Self::Error => LevelFilter::ERROR,
+            Self::Warn => LevelFilter::WARN,
+            Self::Info => LevelFilter::INFO,
+            Self::Debug => LevelFilter::DEBUG,
+            Self::Trace => LevelFilter::TRACE,
+            Self::Off => LevelFilter::OFF,
+        }
+    }
+}
```

Comment:
This could also be a `From`/`Into` impl.

Reply:
I'll switch it over, copied from relay